README.md

```diff
@@ -2,69 +2,82 @@
 
 Copyright Tweag I/O 2024
 
-With `cooked-validators` you can test Cardano smart contracts by writing potentially malicious offchain code.
-You can also use the library to write "normal" offchain code in a comfortable and flexible way.
+`cooked-validators` is a Haskell library to conveniently and
+efficiently write off-chain code for Cardano smart contracts. This
+offchain code will be specifically geared to testing and auditing the
+smart contract in question with further builtin capabilities of the
+library.
 
-In particular, `cooked-validators` helps you:
-- interact with smart contracts written in Plutus (as well as any other language
+In particular, `cooked-validators` allows to:
+- interact with smart contracts written in Plutus or any other language
   that compiles to [UPLC](https://plutonomicon.github.io/plutonomicon/uplc),
   like for example [Plutarch](https://github.com/Plutonomicon/plutarch-plutus)
-  or [Aiken](https://aiken-lang.org/), by loading contracts from byte strings)
-- generate and submit transactions declaratively, while automatically taking
-  care of missing inputs and outputs, balancing, and minimum-Ada constraints
-- construct sequences of transactions in an easy-to-understand abstraction of
-  "the blockchain", which can be instantiated to different actual
-  implementations
-- run sequences of transactions in a simulated blockchain,
-- apply "tweaks" to transactions right before submitting them, where "tweaks"
-  are modifications that are aware of the current state of the simulated
-  blockchain
-- compose and deploy tweaks with flexible idioms inspired by linear temporal
-  logic, in order to turn one sequence of transactions into many sequences that
-  might be useful test cases, generalized in [Graft](https://github.com/tweag/graft)
+  or [Aiken](https://aiken-lang.org/), by loading contracts from byte strings.
+- declaratively define transactions in a high level, type-retaining
+  data structure, and submitting them for validation, while
+  automatically taking care of missing inputs and outputs, balancing,
+  minimum-Ada constraints, collaterals and fees
```

Comment:
I find this sentence a bit convoluted and less readable than the previous one. I would rather keep it and add another item about retaining type information in contexts where they are usually lost.